flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -1759,11 +1759,22 @@ class JobManager(
       case None => None
     }
 
-    // remove all job-related BLOBs from local and HA store
-    libraryCacheManager.unregisterJob(jobID)
-    blobServer.cleanupJob(jobID, removeJobFromStateBackend)
+    // remove all job-related BLOBs from local and HA store, only if the job was removed correctly
+    futureOption match {
+      case Some(future) => future.onComplete{
+        case scala.util.Success(_) => {
+          libraryCacheManager.unregisterJob(jobID)
```

Comment:
This call should also be called if the removal of the job from the `SubmittedJobGraphStore` failed because it does not remove any HA files.